app/src/main/java/com/cornellappdev/android/eatery/MainListFragment.java

```diff
@@ -152,13 +159,13 @@ private void handlePaymentButtonPress(Button button, String payment) {
   public void onClick(View view) {
     switch (view.getId()) {
       case R.id.northButton:
```

Comment:
then you could change this to `handleAreaButtonPress(mButtons.get(view.getId()));`